Std/Data/Queue.lean

```diff
@@ -1,53 +0,0 @@
-/-
```

Comment:
Why is this file being removed?

Reply:
Because it's now in core.

Reply:
I think I asked you about the naming conflict (because it's called `Std.Queue`), but you didn't complain so I assumed that you were okay with it.

Reply:
Oh I see, it's in `Init` now. The docstrings should be relocated there though.

Reply:
"this is only a temporary placeholder" should also probably be removed